### DISCUSSION TITLE
Disposal graph

### DISCUSSION THREAD

**Discussion by jnyrup** [hardkoded/puppeteer-sharp#2117] [Show and tell]
I've struggled a bit with following the flow of disposing a `Browser` so I created a graph of it with the edge weights being the order of execution.

![image](https://user-images.githubusercontent.com/919634/228338127-ab9d7ef5-9e1c-4c01-9d8f-4b4e491d491d.png)

You can remove .txt from this file and upload it to [graphonline](https://graphonline.ru/en/)
[puppeteersharp.graphml.txt](https://github.com/hardkoded/puppeteer-sharp/files/11093215/puppeteersharp.graphml.txt)

Visualizing the depth-first search is then an approximation of the dispose process.
https://user-images.githubusercontent.com/919634/228340100-bb104bef-e698-4b6b-b292-0fbf2143a2ba.mp4

**Comment by kblok** [comment]
That video is great!
